ext/semian/semian.c

```diff
@@ -167,13 +134,11 @@ static VALUE
 update_ticket_count(update_ticket_count_t *tc)
 {
   short delta;
-  struct timespec ts = { 0 };
-  ts.tv_sec = kInternalTimeout;
 
   if (get_max_tickets(tc->sem_id) != tc->tickets) {
     delta = tc->tickets - get_max_tickets(tc->sem_id);
 
-    if (perform_semop(tc->sem_id, kIndexTickets, delta, 0, &ts) == -1) {
+    if (perform_semop(tc->sem_id, kIndexTickets, delta, 0) == -1) {
```

Comment:
I'm not really sure if I like that any better. If we omit `IPC_NOWAIT` then we could risk the process blocking indefinitely *if* there were processes that decided to hold onto their tickets (also seems very unlikely). It's kind of gross, but what about `sleep`ing on a failure and retrying a given number of times (basically emulating what we were doing before with the timeout)?